### PR TITLE
feat: create a new page to do LTI 1.3 migration

### PR DESCRIPTION
**Pull request by PenghaiZhang**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

In the settings page, there is new link which points to the LTI 1.3 migration page.

Please note the migration page has only some placeholder UI in this PR.

image

image

### CODE REVIEW

File: lti13migration.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle. If not, see <http://www.gnu.org/licenses/>.
+ echo "
+  <h2>Update oEQ resource links to use LTI 1.3</h2>
+  <div>
+    <div>
+      Click <a href='?action=download'>here</a> to download the CSV of courses that will be affected by the migration.
```
Comment (edalex-ian) [inline lti13migration.php:20]: I know this is kind of a one use page, however the use of 'click here' is a throw back to early web and generally considered poor for accessibility. Should we perhaps bump upgrade this page a bit and use buttons?

For example:

```html
<p>
  Before starting the migration, you can download a CSV which will list all courses which will
  be updated. This CSV can be used for review, and no system modifications will occur.
<p>
<form action="?action=download" method="GET">
  <button>Download CSV</button>
</form>
```

I know it's onerous to need a whole form, but the other options around buttons and links are also not really aligned with accessibility.

If this approach is followed, the same should be done for the migration link below.

**Review by edalex-ian — COMMENTED** [review]
Thanks @PenghaiZhang . Have a couple of suggestions and a query for you.

File: settings.php [hunk 2]
```diff
@@ -105,4 +105,8 @@ function ecs($configoption, $params = null) {
     $intercepttype = new admin_setting_configselect('equella_intercept_files', get_string('interceptfiles', 'equella'), get_string('interceptfilesintro', 'equella'), 0, $choices);
 
     $settings->add($intercepttype);
+
+    $settings->add(new admin_setting_heading('equella_lti_migration', ecs('lti13.migration.title'), ''));
```
Comment (edalex-ian) [inline settings.php:109]: One other quick comment. Note how the sections above have a comment to highlight which part of the settings they below too (e.g. Drag and drop). Could you please add such a comment for this new section too?

File: lang/en/equella.php [hunk 3]
```diff
@@ -101,6 +101,11 @@
 $string['config.lti.secret.help'] = 'Client secret is required if LTI is enabled.';
 $string['config.lti.liscallback.title'] = 'Outcome callback URL';
 
+////////////////////////////////////////////////////////
+// LTI 1.3 migration
+$string['config.lti13.migration.title'] = 'LTI 1.3 Migration';
+$string['config.lti13.migration.description'] = 'Migrate openEquella resources to use LTI 1.3';
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply lang/en/equella.php:107]: It should be `openEQUELLA` :smile: .

File: lti13migration.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle. If not, see <http://www.gnu.org/licenses/>.
+ echo "
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply lti13migration.php:16]: No, it's not encased in anywhere. It should just be a independent HTMLdocument.

And The `out()` only returns the URL string.  

I will add `head, body etc`.

File: settings.php (hunk 2 above)
Comment (PenghaiZhang) [reply settings.php:109]: Too easy. I will add the comment.

File: lti13migration.php (hunk 1 above)
Comment (PenghaiZhang) [reply lti13migration.php:20]: Yeah let's make this page a bit better. I will use `form` .

File: lti13migration.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (edalex-ian) [reply lti13migration.php:17]: This all looks much better! :+1: 

But, I note you still didn't do `<head>`, `<html>` or `<doctype>` etc.

Also, in this case you don't need to use `echo`. Being a PHP file you can just place the HTML file directly in there. You only need `echo` if you want to do something dynamic.

**Review by edalex-ian — COMMENTED** [review]
Great @PenghaiZhang , almost there - just also need Cath to confirm content.

Also, just one more look at the new page. BTY.

File: lti13migration.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply lti13migration.php:17]: Yes I do want t do something dynamic . When the request of migration is submitted, the page will show another paragraph to say something like `A task has been created. Please go to xxx to view the task result`.

**Review by edalex-ian — APPROVED** [review]
Excellent, thanks @PenghaiZhang :+1:

**Pull request merged**